eosc_pl/transform/utils/loader.py

```diff
@@ -2,37 +2,55 @@
 """Load data"""
 import logging
 import urllib.parse
+from typing import Optional
 
 import pandas as pd
 import requests
-
-from eosc_pl.transform.utils.validate import validate_loaded_pd_df
+from transform.utils.consts import Repository
+from transform.utils.validate import validate_loaded_pd_df
 
 logger = logging.getLogger(__name__)
 
 
-def pd_load_datasets(url: str) -> pd.DataFrame | None:
+def _get_params(per_page: int, start: int, repository: Repository) -> str:
```

Comment:
that function creates another dependency when you want to add a new repository. Not a fan of that. Defining URLs in settigns.py should be enough

Reply:
`start` parameter is changing on each function call (each page)
Moving fstring to a function makes python re-evaluate it each time